infra/Pos/Diffusion/Subscription/Common.hs

```diff
@@ -238,15 +206,57 @@ networkSubscribeTo' logTrace oq bucket nodeType peersVar keepalive status sendAc
     msgSubscriptionTerminated :: NodeId -> SubscriptionTerminationReason -> Text
     msgSubscriptionTerminated = sformat ("subscriptionWorker: lost connection to "%shown%" "%shown)
 
-    alterPeerSubStatus :: NodeId -> Maybe SubscriptionStatus -> IO ()
-    alterPeerSubStatus peer s = atomically $ do
+    alterPeerSubStatus :: Maybe SubscriptionStatus -> IO ()
+    alterPeerSubStatus s = atomically $ do
         stats <- readTVar status
         let !stats' = Map.alter fn peer stats
         writeTVar status stats'
         where
             fn :: Maybe SubscriptionStatus -> Maybe SubscriptionStatus
             fn x = getOption (Option x <> Option s)
```

Comment:
I wonder if we're using the right monoid here, `Option Nothing` is a unit for `Option` so `alterPeerSubStatus Nothing` will not remove the peer from the `TVar`.

Reply:
Oh right this is per-peer, so assuming there's only one writer per-peer then we can just overwrite the old status.